lib/pf/action.pm

```diff
@@ -192,24 +193,21 @@ sub action_execute {
     my $logger = Log::Log4perl::get_logger('pf::action');
     my $leave_open = 0;
     my @actions = class_view_actions($vid);
-    # Sort the actions in reverse order in order to always finish with the autoreg action
     @actions = sort { $b->{action} cmp $a->{action} } @actions;
     foreach my $row (@actions) {
         my $action = lc $row->{'action'};
         $logger->info("executing action '$action' on class $vid");
-        if ( $action eq $TRAP ) {
+        if ( $action eq $REEVALUATE_ACCESS ) {
             $leave_open = 1;
-            action_trap( $mac, $vid );
-        } elsif ( $action eq $EMAIL ) {
-            action_email( $mac, $vid, $notes );
+            action_reevaluate_access( $mac, $vid );
+        } elsif ( $action eq $EMAIL_ADMIN ) {
+            action_email_admin( $mac, $vid, $notes );
+        } elsif ( $action eq $EMAIL_USER ) {
+            action_email_user( $mac, $vid, $notes );
         } elsif ( $action eq $LOG ) {
             action_log( $mac, $vid );
         } elsif ( $action eq $EXTERNAL ) {
             action_api( $mac, $vid );
-        } elsif ( $action eq $WINPOPUP ) {
-            action_winpopup( $mac, $vid );
-        } elsif ( $action eq $AUTOREG ) {
```

Comment:
Without the dhcp fingerprint ?
I'll assume you meant with.

During the lunch & learn we discussed it should now be using VLAN filters as violations should be actual violations and not a way to grant access to the network.